examples/stable_diffusion_v2/tools/safety_checker/safety_checker.py

```diff
@@ -0,0 +1,271 @@
+"""
+Implementation of two versions of safety checker in stable
+diffusion 1/2, respectively
+"""
+import argparse
+import os
+import sys
+
+import yaml
+
+# equivalent to no-check-certificate flag in wget
+os.environ["CURL_CA_BUNDLE"] = ""
+# add current working dir to path to prevent ModuleNotFoundError
+sys.path.insert(0, os.getcwd())
+
+from tools._common import L2_norm_ops, load_images
+from tools._common.clip import CLIPImageProcessor, CLIPTokenizer, parse
+from tools.safety_checker.utils import locate_model
+
+import mindspore as ms
+from mindspore import ops
+
+
+class SafetyChecker:
+    def __init__(
+        self,
+        safety_version=2,
+        backend="ms",
+        config="tools/_common/clip/configs/clip_vit_l_14.yaml",
+        ckpt_path=None,
+        tokenizer_path="ldm/models/clip/bpe_simple_vocab_16e6.txt.gz",
+        model_name="openai/clip-vit-large-patch14",
+        settings_path="tools/safety_checker/safety_settings_v2.yaml",
+        threshold=0.2,
+        **kwargs,
+    ):
+        assert safety_version in [1, 2]
+
+        if backend == "pt":
+            from transformers import CLIPModel, CLIPProcessor
+
+            model = CLIPModel.from_pretrained(model_name)
+            processor = CLIPProcessor.from_pretrained(model_name)
+
+            def process_text(text):
+                return processor(text=text, return_tensors="pt", padding=True).input_ids
+
+            if safety_version == 2:
+                import torch
+                from tools.safety_checker.nsfw_model_pt import NSFWModelPT
+
+                nsfw_model = NSFWModelPT()
+                model_path = locate_model(backend="pt")
+                assert model_path is not None
+                nsfw_model.load_state_dict(torch.load(model_path))
+
+        elif backend == "ms":
+            # parse config file
+            from tools._common.clip import CLIPModel
+
+            config = parse(config, ckpt_path)
+            self.image_size = config.vision_config.image_size
+            self.dtype = ms.float32 if config.dtype == "float32" else ms.float16
+            model = CLIPModel(config)
+            processor = CLIPImageProcessor()
+            tokenizer = CLIPTokenizer(tokenizer_path, pad_token="!")
+
+            def process_text(text):
+                return ms.Tensor(tokenizer(text, padding="max_length", max_length=77)["input_ids"])
+
+            if safety_version == 2:
+                from tools.safety_checker.nsfw_model import NSFWModel
```

Comment:
suggest to put `import` things on the top of the file. May not be a good practice to put them in the content

Reply:
fixed